File: src/BaseHmiTypes/Screens/Shapes/HmiCentricShapeBase.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public abstract class HmiCentricShapeBase : HmiShapeBase
{
    public HmiProperty<double> CenterX { get; set; } = 0;

    public HmiProperty<double> CenterY { get; set; } = 0;
}

File: typescript/base-hmi-types/src/screens/shapes/HmiCentricShapeBase.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiShapeBase } from "./HmiShapeBase.js";

export abstract class HmiCentricShapeBase extends HmiShapeBase {
  centerX: HmiProperty<number> = staticProperty(0);
  centerY: HmiProperty<number> = staticProperty(0);
}
